### PR TITLE
DOCS: Add trigger to build docsite when changes are merged to master

### PR DESCRIPTION
**Pull request by tkfu**

### CODE REVIEW

**Comment by tkfu** [comment]
> It's better to use separate stage to make sure those jobs are self-contained and independent.

Well, we generally _want_ jobs to run in parallel as much as reasonably possible, and we want to group them by function. All this job does is fire off a trigger to run another pipeline; there's no possible way for it to interfere with any other job in the pipeline. (Or for any other job to interfere with it.)

That said, the veracode check isn't triggering a pipeline, it's running a test, so it should probably go in a stage called 'test' rather than a stage called 'trigger'--I'll make that change.

**Comment by lbonn** [comment]
@tkfu the actual veracode test is running in some other CI, so that's why I've originally asked @zabbal to group it with our other triggers (in aktualizr originally), even if it's not technically a gitlab trigger job.

**Comment by tkfu** [comment]
Ah, that makes sense. @zabbal, alright with you if I revert? I don't want to end up with a CI file that has a stage for every job, and this is very clearly a trigger.

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -20,3 +21,14 @@ trigger-veracode-scan:
...
+trigger-docsite-build:
+  stage: publish-docs-if-changed
+  only:
+    refs:
+      - master
+    changes:
+      - ota-plus-web/docs/**/*
+  trigger:
+    project: olp/edge/ota/documentation/ota-connect-docs
+    branch: master
```
Comment (jochenschneider) [inline .gitlab-ci.yml]: I would assume this is the default.

**Pull request merged**